### PR TITLE
ENH: scipy.sparse: copy keyword added to asformat method

### PR DESCRIPTION
**Pull request by jor-**
I added a `copy` keyword to the `asformat` method of the sparse matrices in `scipy.sparse`.

### CODE REVIEW

File: scipy/sparse/base.py [hunk 1]
```diff
@@ -296,25 +296,34 @@ def __len__(self):
...
-            return self
+            if copy:
+                return self.copy()
+            else:
+                return self
         else:
-            return getattr(self, 'to' + format)()
+            try:
+                convert_method = getattr(self, 'to' + format)
+            except AttributeError as e:
```
Comment (perimosocordiae) [inline scipy/sparse/base.py]: Don't need to bind the error to `e` here.

**Review by perimosocordiae — APPROVED** [review]
Makes sense to have this option here, too.

File: scipy/sparse/base.py [hunk 2]
```diff
@@ -296,25 +296,34 @@ def __len__(self):
...
-        format : {string, None}
-            desired sparse matrix format
-                - None for no format conversion
-                - "csr" for csr_matrix format
-                - "csc" for csc_matrix format
-                - "lil" for lil_matrix format
-                - "dok" for dok_matrix format and so on
+        format : {str, None}
+            The desired sparse matrix format ("csr", "csc", "lil", "dok", ...)
+            or None for no conversation.
```
Comment (crayxt) [inline scipy/sparse/base.py]: `no conversation` -> `no conversion`?

**Comment by jor-** [comment]
Thanks for the comments. I made the proposed changes.

**Comment by perimosocordiae** [comment]
Looks good, merging.

**Pull request merged**